src/Components/DesignAdjustments.tsx

```diff
@@ -44,27 +43,52 @@ export const DesignAdjustments = connect(
           <style type="text/css">{`:root{\n  ${styles.join('\n  ')}\n}`}</style>
         </Helmet>
 
-        <CustomFonts root={root} />
+        {root.get('siteFontHeadline').map((font) => (
+          <FontFace
+            font={font}
+            fontFamily={headlineFontFamily}
+            key={`custom-headline-font-family${font.id()}`}
+          />
+        ))}
+
+        {root.get('siteFontBody').map((font) => (
+          <FontFace
+            font={font}
+            fontFamily={bodyFontFamily}
+            key={`custom-headline-font-family${font.id()}`}
+          />
+        ))}
+
         {children}
       </>
     )
   },
   { loading: () => null },
 )
 
-const CustomFonts = connect(function CustomFonts({
-  root,
-}: {
-  root: HomepageInstance
-}) {
-  return (
-    <>
-      {uniqBy(
-        [...root.get('siteFontHeadline'), ...root.get('siteFontBody')],
-        (obj) => obj.id(),
-      ).map((font) => (
-        <FontComponent page={font} key={font.id()} />
-      ))}
-    </>
-  )
-})
+const FontFace = connect(
+  function FontFace({ font, fontFamily }: { font: Obj; fontFamily: string }) {
+    if (!isFont(font)) return ''
```

Comment:
* [ ] FYI `return ''` looks strange, especially close to the `loading` component. How about `return null`?